### PR TITLE
restore: retry if deliver KVs to importer

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

```
2019/04/28 07:50:23.039  [error] [`db1903_baofu`.`cm_entry`:11] [/data1/hybrid/baofu/db1903_baofu.cm_entry.3771.sql:0] error EOF
2019/04/28 07:50:23.039  [info] [`db1903_baofu`.`cm_entry`:11] encode kv data and write takes 1m11.606816959s (read 22382456589, written 392441556)
2019/04/28 07:50:23.039  [error] [`db1903_baofu`.`cm_entry`:11] error EOF
```

### What is changed and how it works?

Retry if deliver KVs to the importer failed.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

Related changes

### CODE REVIEW

**Comment by lonng** [comment]
@kennytm @july2993 PTAL

File: lightning/restore/restore.go [hunk 1]
```diff
@@ -1426,6 +1432,13 @@ func (cr *chunkRestore) deliverLoop(
 	var channelClosed bool
 	var dataKVs, indexKVs []kvenc.KvPair
 
+	deliverTask := t.logger.With(
+		zap.Int32("engineNumber", engineID),
+		zap.Int("fileIndex", cr.index),
+		zap.Stringer("path", &cr.chunk.Key),
+		zap.String("task", "deliver"),
+	).Begin(zap.InfoLevel, "deliver started")
```
Comment (kennytm) [inline lightning/restore/restore.go]: What's this?

Comment (lonng) [reply lightning/restore/restore.go]: Logging detail information if the write to engine error occurred.

Comment (kennytm) [reply lightning/restore/restore.go]: You didn't call `deliverTask.End()` though, so using the result of `With()` as a logger is sufficient.

Comment (lonng) [reply lightning/restore/restore.go]: addressed

**Pull request merged**